### ISSUE TITLE
[Feature]: Improve GitHub Workflow to use Composition

### ISSUE THREAD

**Issue by seriouslysean** [cuttle-cards/cuttle#129]
### Feature Summary

Remove `run-npm-script` and replace with composition

### Detailed Description

See https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/ for more information

**Comment by itsalaidbacklife** [comment]
@seriouslysean would you say this has been accomplished with our current gh actions setup?

**Comment by seriouslysean** [comment]
I don't think so, there's probably better caching we can do with named builds, I'd leave this open imo if there's someone pretty passionate about docker.
